components/description-list/description-list-wrapper.js

```diff
@@ -0,0 +1,97 @@
+import { bodyCompactStylize, labelStylize } from '../typography/styles.js';
+import { css, html, LitElement } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const descriptionListStyles = [
+	labelStylize('dt'),
+	bodyCompactStylize('dd'),
+	css`
+		:host {
+			--d2l-dt-min-width: min-content;
+			--d2l-dt-max-width: 10rem;
+			--d2l-dd-min-width: 50%;
+		}
+		dl {
+			align-items: baseline;
+			display: var(--d2l-description-list-dl-display, grid);
+			gap: 0.5rem 1.5rem;
+			grid-auto-flow: row;
+			grid-template-columns: minmax(var(--d2l-dt-min-width), auto) minmax(var(--d2l-dd-min-width), 1fr);
+		}
+		dt {
+			max-width: var(--d2l-dt-max-width);
+		}
+		dd {
+			margin: var(--d2l-description-list-dd-margin, 0);
+		}
+	`,
+];
+
+/**
+ * Wraps a native <dl> element, providing styling and resize behavior.
+ * @slot - Content to wrap
+ */
+class DescriptionListWrapper extends LitElement {
+	static get properties() {
+		return {
+			breakpoint: { type: Number, reflect: true },
+			_stacked: { state: true },
+
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.stacked {
+				--d2l-description-list-dl-display: block;
+				--d2l-dt-max-width: unset;
+				--d2l-description-list-dd-margin: 0 0 0.5rem 0;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this._stacked = false;
+		this.breakpoint = 350;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	render() {
+		const classes = {
+			'stacked': this._stacked,
+		};
+		return html`<slot class="${classMap(classes)}"></slot>`;
+	}
+
+	_checkIfShouldStack() {
+		if (this.clientWidth < this.breakpoint && !this._stacked) {
+			this._stacked = true;
+		} else if (this.clientWidth >= this.breakpoint && this._stacked) {
+			this._stacked = false;
+		}
+	}
+
+	_onResize() {
```

Comment:
I know this is a WIP but figure I thought would inquire anyway - any chance we can use the [ResizeObserverEntry](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry) that the `ResizeObserver` provides to the handler? Using those values would always be preferred if possible, as opposed to potentially forcing a layout by calling `this.clientWidth` above. After all, that's one of the advantages of using RO. 

If trying this... be aware that we're limited to using [contentRect](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/contentRect) since `borderBoxSize` and `contentBoxSize` were not added to the entry API until much later, and so the RO polyfill does not support it. The Mozilla doc shows how it can be detected and adapted, but we haven't bothered so far.

Reply:
Tried this out in 31deda23fb5fb35fd617959fe3c3af2f932f43dc - looks like it works!

I'm open to feedback on the best way to implement `ResizeObserverEntry`. I based my implementation off this:

https://github.com/BrightspaceUI/core/blob/ce0150cabc808de416cd0a9e1f77f64038d28c15/components/empty-state/empty-state-illustrated.js#L120-L125

Reply:
Looks great!